webapp/src/components/TodoList.vue

```diff
@@ -0,0 +1,215 @@
+      <!--      </div>-->
+    </div>
+    <ol>
+      <TodoItem
+        v-for="todo in this.todoListData"
+        :todo="todo"
+        :edit-mode="initialEditMode"
+        :key="todo.id"
+        @delete-entry="deleteEntry"
+        @save-entry="saveEntry"
+        @remove-entry-local="removeEntryLocal"
+      />
+    </ol>
+  </div>
+</template>
+
+<script>
+import gql from "graphql-tag";
+import TodoItem from "./TodoItem";
+
+export default {
+  name: "TodoList",
+  components: { TodoItem },
+  props: ["user"],
+  apollo: {
+    todoListData: {
+      query: gql`
+        query {
+          todoListData: todos {
+            id
+            text
+            author {
+              name
+            }
+          }
+        }
+      `
+    }
+  },
+  data() {
+    return {
+      initialEditMode: false,
+      isEditingAuthor: false
+    };
+  },
+  methods: {
+    removeEntryLocal() {
+      this.todoListData.splice(this.todoListData.length - 1, 1);
+    },
+    addEntry() {
+      this.initialEditMode = true;
+      const newTodo = {
+        message: "New Todo",
+        author: {
+          name: this.user.username
+        }
+      };
+      this.todoListData.push(newTodo);
+    },
+    deleteEntry: function(todo) {
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation delToDo($id: ID) {
+              delToDo(id: $id) {
+                id
+                text
+                author {
+                  name
+                }
+              }
+            }
+          `,
+          variables: {
+            id: todo.id
+          }
+        })
+        .then(body => {
+          this.todoListData = body.data.delToDo.slice();
+        });
+    },
+    saveEntry: function(todo) {
+      if (todo.id == null) {
+        this.initialEditMode = false;
+        this.$apollo
+          .mutate({
+            mutation: gql`
+              mutation addToDo($text: String!, $authorName: String!) {
+                addToDo(text: $text, authorName: $authorName) {
+                  id
+                  text
+                  author {
+                    name
+                  }
+                }
+              }
+            `,
+            variables: {
+              text: todo.text,
+              authorName: this.user.username
+            }
+          })
+          .then(body => {
+            this.todoListData = body.data.addToDo.slice();
+          });
+      } else {
+        this.$apollo
+          .mutate({
+            mutation: gql`
+              mutation updateToDo(
+                $id: ID!
+                $text: String!
+                $authorName: String!
+              ) {
+                updateToDo(id: $id, text: $text, authorName: $authorName) {
+                  id
+                  text
+                  author {
+                    name
+                  }
+                }
+              }
+            `,
+            variables: {
```

Comment:
It doesn't hurt to add `id` as unused variable for the if-branch of creating the todo.

So you can DRY your code like this:
```js
const mutation = !todo.id ? createMutation : updateMutation
const { id, text } = todo
const variables = { id, text, authorName: this.user.username }
const data = this.$apollo.mutate({mutation, variables})
```

Reply:
I don't like that. using ```todo.id``` makes it easier to identify where the ```id``` comes from.
Its not always about clean code. Readable code is important too 😉